app/src/main/java/com/academy/ramon/marvelcomicviewer/MainActivity.java

```diff
@@ -49,11 +54,13 @@ public void getHeroesList(){
             public void onResponse(Call<HeroesResponse> call, Response<HeroesResponse> response) {
                 HeroesResponse heroesResponse;
                 heroesResponse = response.body();
-                rvHeroes.setAdapter(new HeroAdapter(heroesResponse.getItems(),rvHeroes.getContext()));
-                heroes = heroesResponse.getItems();
-                Log.d("hey there", response.body().toString());
-            }
+                heroes = heroesResponse.getData().getResults();
+                HeroAdapter adapter = new HeroAdapter(heroes, rvHeroes.getContext());
```

Comment:
I dont fully understand this, you mean: Save the context inside the adapter and then take it from there? Maybe this is why Im getting this error:  E/RecyclerView: No adapter attached; skipping layout Eventhough the app works fine (so far) and the adapter is set properly